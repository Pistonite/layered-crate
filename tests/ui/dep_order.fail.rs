use layered_crate::layers;

#[layers]
mod src {
    #[depends_on(y)]
    #[depends_on(x)]
    pub extern crate z;

    pub extern crate x;
    pub extern crate y;
}

fn main() {
    x::do_something();
    z::zzz();
    y::do_something();
}
