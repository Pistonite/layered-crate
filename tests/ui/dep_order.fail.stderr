error: #[depends_on(x)] should be before #[depends_on(y)] to ensure consistent order of modules
 --> tests/ui/dep_order.fail.rs:6:18
  |
6 |     #[depends_on(x)]
  |                  ^

warning: unused import: `x`
 --> tests/ui/dep_order.fail.rs:6:18
  |
6 |     #[depends_on(x)]
  |                  ^
  |
  = note: `#[warn(unused_imports)]` on by default
